src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -163,13 +172,13 @@ public IEnumerable<string> GetCustomDefaultConfigurationValueIfSpecified()
             if (string.Equals(Environment.GetEnvironmentVariable(EnvironmentVariableNames.DOTNET_CLI_LAZY_PUBLISH_AND_PACK_RELEASE_FOR_SOLUTIONS), "true", StringComparison.OrdinalIgnoreCase))
             {
                 // Evaluate only one project for speed if this environment variable is used. Will break more customers if enabled (adding 8.0 project to SLN with other project TFMs with no Publish or PackRelease.)
-                return GetSingleProjectFromSolution(sln, globalProps);
+                return GetSingleProjectFromSolution(sln, slnPath, globalProps);
             }
 
-            Parallel.ForEach(sln.Projects.AsEnumerable(), (project, state) =>
+            Parallel.ForEach(sln.SolutionProjects.AsEnumerable(), (project, state) =>
             {
 #pragma warning disable CS8604 // Possible null reference argument.
-                string projectFullPath = Path.Combine(Path.GetDirectoryName(sln.FullPath), project.FilePath);
+                string projectFullPath = Path.Combine(Path.GetDirectoryName(slnPath), project.FilePath);
```

Comment:
Why does this method (in code that was from before this PR) do a Parallel.ForEach on a bunch of projects when you just want one? Seems wasteful